### DISCUSSION TITLE
How to build an app on top of any-sync? Is there any guide?

### DISCUSSION THREAD

**Discussion by Slaaavo** [anyproto/contributors#140] [Q&A] [answered]
As the title suggests, I’d like to build an app on top of the any-sync protocol, which I found out is open-source and available. I only found this documentation page [Document 1](https://tech.anytype.io/any-sync/overview) which gives a good overview of the principles on which the protocol is built. However, I haven’t found any documentation on how I might be able to use it in the app I’m building. Is this available anywhere? Can I get some pointers at least? :slight_smile:

**Comment by mcrakhman** [comment, marked as answer]
Hi! No unfortunately we don't have anything like that at the moment, but you can get the gist by looking at the code of `any-sync-node` repo (check `commonspace` in `any-sync` and `nodespace` in `any-sync-node`, you can also check the `cmd` dir in `any-sync-node` to see how the whole thing is wired up). 

To be honest it would require some effort to fully describe how stuff works, but essentially you need to provide some components for the `SpaceService` to be working (e.g. cache for objects, peer pools, storage etc). As soon as you provide them, you would be able to create spaces and subsequently create objects. The changes which you do will be internally (via these components) propagated to the designated peers. 

Btw what app do you want to build, so maybe I can give a few more details?

**Reply by njouanin** [reply]
Hi, is the process of building an app on top of any-sync more documented 2 years later ?

**Comment by Slaaavo** [comment]
Thanks for the pointers. I was wrongly looking through `any-heart` repo for usage.

I want to build an app for tracking of personal finances. They suffer the same issue as note taking apps, either cloud based at the loss of privacy or poor sync capabilities.

At first, I'd like to do it without a network of sync nodes and rely solely on p2p sync in case the user (which will be mainly me at first) wants to sync between devices, i. e. sync directly from phone to phone without intermediary, or maybe set up a node on my RPi, that would do everything.

I'd just like to know if there are any constraints on the technologies I can use if I want to use any-sync. I'm thinking about a simple local first flutter app with SQLite db.

**Reply by sashailinsky** [reply]
It will be cool if you can integrate Zenmoney
